.isort.cfg

```diff
@@ -0,0 +1,10 @@
+# A Black (https://github.com/psf/black)-compatible isort
+# (https://timothycrosley.github.io/isort/) config. Copy-pasted from Black's
+# README.
+[settings]
+multi_line_output=3
+include_trailing_comma=True
+force_grid_wrap=0
+use_parentheses=True
+line_length=88
+known_third_party=dateutil,factory,faker,hypothesis,pytest,slugify,webtest,zope
```

Comment:
Hmm ... it seems surprising that we'd need to list all of these here given that third party packages can be identified by install location (the `site-packages` dir). I wonder why isort doesn't do that?

Reply:
Yeah, we don't want to get in the habit of having to list all the dependencies again. I wonder why these ones specifically?

Reply:
It doesn't get things from site-packages because isort is run in its own separate virtualenv (created by tox) where all the requirements are not installed in that virtualenv's site-packages.

We could add isort to the test venv where all the dev and test deps are installed, and that might fix this, but then you would have to create the massive test venv (installing all dev and test deps), waiting for that to complete just to run `isort --check` and find out that you've got an import out of order.

It does create this small amount of maintenance and duplication in isort's case, but running tools like flake8, black and isort in their own venv's has a bunch of advantages:

* It's much faster both in dev and on CI to install just isort and run it. If CI is gonna fail because of import sorting, you'll find out sooner, without waiting for everything to install first.
* isort can run in parallel with the tests, black and linting as they don't depend on waiting for the deps to be installed in the same venv.
* It avoids any potential problems with dependency conflicts between isort and our dev and test deps, and other tools like black and flake8
* It allows isort to use a different version of Python than the devserver, tests, and others tools do if it needs to. For example we run Black for Via, but Black needs Python 3.6 whereas Via is Python 2.7, but that's not a problem because we run Black in a separate venv. Similarly, if we want to upgrade a project to Python 3.8 one day, but isort, or Black, or some other tool we're using, doesn't run in 3.8 yet, that won't be a problem. We only need our actual production and test deps to support Python 3.X and then we can upgrade to it.

Anyway, so given that isort is reading the third-party requirements from the requirements files not from site-packages, certain package names confuse it:

* dateutil is called python-dateutil in requirements.txt (i.e. the package install name differs from the import name)
* Similarly factory is factory-boy
* faker is called Faker (capital F) on PyPI
* slugify is python-slugify
* We don't have zope as a dep but we have zope.deprecation, zope.interface, and zope.sqlalchemy, which are packages that provide things within the zope namespace
* pytest, hypothesis and webtest are listed in tox.ini rather than in a requirements file, that might be the problem with them. Extracting these out into `requirements-test.txt` and `requirements-functests.txt` might fix these.